rlbot_gui/story/story_challenge_setup.py

```diff
@@ -0,0 +1,462 @@
+from datetime import datetime
+from os import path
+from typing import List, Tuple
+
+import json
+import random
+import time
+import traceback
+
+from rlbot.utils.game_state_util import GameState
+from rlbot.utils.structures.game_data_struct import GameTickPacket, Vector3
+from rlbot.parsing.match_settings_config_parser import (
+    game_mode_types,
+    boost_amount_mutator_types,
+    map_types,
+    max_score_types,
+    rumble_mutator_types,
+    match_length_types,
+)
+from rlbot.matchconfig.match_config import (
+    PlayerConfig,
+    MatchConfig,
+    MutatorConfig,
+    Team,
+)
+from rlbot.setup_manager import SetupManager
+from rlbot_gui import gui as rlbot_gui  # TODO: Need to remove circular import
+
+from rlbot_gui.story.load_story_descriptions import BOTS_CONFIG
+
+WITNESS_ID = random.randint(0, 1e5)
+
+DEBUG_MODE_SHORT_GAMES = False
+
+def setup_failure_freeplay(setup_manager: SetupManager, message: str, color_key="red"):
```

Comment:
This is just a clever way to render on screen, but not allow the bots to do so during the match, right?

Another, somewhat clunky way to do so would be to simulate a key press of "Page Up", but I don't think that has any real advantages. I think I prefer the way you did it, but I wanted to make you aware of the option in case you disagree.

Reply:
That is a cool idea but yea I agree with you, this is simpler for now.